itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java

```diff
@@ -138,6 +141,211 @@ public void tearDown() throws Throwable {
+    assertFalse(fs.exists(new Path(dumpPath, ReplAck.FAILOVER_READY_MARKER.toString())));
+    assertFalse(MetaStoreUtils.isDbBeingFailedOver(primary.getDatabase(primaryDbName)));
+
+    replica.load(replicatedDbName, primaryDbName)
+            .run("use " + replicatedDbName)
+            .run("show tables")
+            .verifyResults(new String[]{"t1", "t2"})
+            .run("repl status " + replicatedDbName)
+            .verifyResult(dumpData.lastReplicationId);
+
+    primary.run("use " + primaryDbName)
+            .run("insert into t1 values(1)")
+            .run("insert into t2 partition(name='Bob') values(11)")
+            .run("insert into t2 partition(name='Carl') values(10)");
+
+    FailoverMetaData failoverMD = new FailoverMetaData(dumpPath, conf);
+    Long cursorPoint = primary.getCurrentNotificationEventId().getEventId();
+    failoverMD.setCursorPoint(cursorPoint);
+    failoverMD.setOpenTxns(Arrays.asList(1L, 2L, 3L));
+    failoverMD.setAbortedTxns(new ArrayList<>());
+    Long failoverEventId = primary.getCurrentNotificationEventId().getEventId();
+    failoverMD.setEventId(failoverEventId);
+    failoverMD.setTxnsWithoutLocks(new ArrayList<>());
+    failoverMD.write();
+    assertTrue(fs.exists(new Path(dumpPath, FailoverMetaData.FAILOVER_METADATA)));
+
+    dumpData = primary.run("insert into t1 values(205)")
+            .run("insert into t2 partition(name='Bob') values(205)")
+            .dump(primaryDbName, failoverConfigs);
+
+    fs = new Path(dumpData.dumpLocation).getFileSystem(conf);
+    dumpPath = new Path(dumpData.dumpLocation, ReplUtils.REPL_HIVE_BASE_DIR);
+    assertTrue(fs.exists(new Path(dumpPath, DUMP_ACKNOWLEDGEMENT.toString())));
+    assertTrue(fs.exists(new Path(dumpPath, FailoverMetaData.FAILOVER_METADATA)));
+    assertTrue(fs.exists(new Path(dumpPath, ReplAck.FAILOVER_READY_MARKER.toString())));
+    assertTrue(MetaStoreUtils.isDbBeingFailedOver(primary.getDatabase(primaryDbName)));
+    assertTrue(failoverEventId >= Long.parseLong(dumpData.lastReplicationId));
+    failoverMD = new FailoverMetaData(dumpPath, conf);
+
+    assertTrue(cursorPoint.equals(failoverMD.getCursorPoint()));
+    assertTrue(failoverEventId.equals(failoverMD.getEventId()));
+    assertTrue(failoverMD.getAbortedTxns().isEmpty());
+    assertTrue(failoverMD.getOpenTxns().equals(Arrays.asList(1L, 2L, 3L)));
+
+    replica.load(replicatedDbName, primaryDbName)
+            .run("use " + replicatedDbName)
+            .run("show tables")
+            .verifyResults(new String[]{"t1", "t2"})
+            .run("repl status " + replicatedDbName)
+            .verifyResult(dumpData.lastReplicationId)
+            .run("select id from t1")
+            .verifyResults(new String[]{"1"})
+            .run("select rank from t2 order by rank")
+            .verifyResults(new String[]{"10", "11"});
+
+    assertTrue(fs.exists(new Path(dumpPath, ReplAck.LOAD_ACKNOWLEDGEMENT.toString())));
+  }
+
```

Comment:
What happens if everything is done but ACK is not creeated? Can we include that as well?